File: output/GPT3.5-Turbo/xCodeEval_remove_4000/C++/Python/e1a47493d2e2ed8b3061f6d87598cbd8.py
import sys
from typing import List, Tuple

def solve() -> int:
    n, k = map(int, input().split())
    k += 1
    a = list(map(int, input().split()))
    a.sort(reverse=True)
    b = [0] * (n + 1)
    for i in range(n):
        b[n - 1 - i] = a[n - 1 - i] * (i // max(k, 1))
    for i in range(n - 1, -1, -1):
        b[i] += b[i + 1]
    cum = 0
    s = 0
    ans = b[0]
    if k > 0:
        for i in range(n):
            res = s + b[i]
            longest = (n - i - 1) // k + 1
            ans = max(ans, res + cum * longest)
            s += cum
            cum += a[i]
    print(ans)
    return 0

if __name__ == "__main__":
    solve()

File: output/GPT3.5-Turbo/xCodeEval_remove_4000/C++/Go/e1a47493d2e2ed8b3061f6d87598cbd8.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Split(bufio.ScanWords)

	n := nextInt(scanner)
	k := nextInt(scanner)
	k++

	a := make([]int, n)
	for i := 0; i < n; i++ {
		a[i] = nextInt(scanner)
	}

	sort.Sort(sort.Reverse(sort.IntSlice(a)))

	b := make([]int64, n)
	for i := 0; i < n; i++ {
		b[n-1-i] = int64(a[n-1-i]) * int64(i/max(k, 1))
	}

	for i := n - 1; i >= 0; i-- {
		b[i] += b[i+1]
	}

	cum := int64(0)
	sum := int64(0)
	ans := b[0]

	if k > 0 {
		for i := 0; i < n; i++ {
			res := sum + b[i]
			longest := (n - i - 1) / k + 1
			ans = max(ans, res+cum*int64(longest))
			sum += cum
			cum += int64(a[i])
		}
	}

	fmt.Println(ans)
}

func nextInt(scanner *bufio.Scanner) int {
	scanner.Scan()
	num, _ := strconv.Atoi(scanner.Text())
	return num
}

func max(a, b int64) int64 {
	if a > b {
		return a
	}
	return b
}
